### PR TITLE
Add back/forward buttons to (blog) articles

### LINKED ISSUE #5809: Implement navigation buttons at the end of the pages
Implementation follow-up for https://github.com/GoogleChrome/developer.chrome.com/issues/5633. Carefully consider where this feature makes sense apart from blog articles.

### PR DESCRIPTION
**Pull request by simgobb**
Fixes #5809

Changes proposed in this pull request:

- Added Next/Prev buttons at the end of articles and blog posts
- Added filters to handle content types
- Added collections for articles and blog posts

### CODE REVIEW

File: site/_includes/partials/post-navigation.njk [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  {% set contentItems = collections.articles %}
+{% endif %}
+
+{% if page.url | getContentType === 'blog' %}
+  {% set contentItems = collections.blogPosts %}
+{% endif %}
+
+{% set currentPostIndex = contentItems | indexOfPostByUrl(page.url) %}
+{% set previousPost = contentItems[currentPostIndex - 1] %}
+{% set nextPost = contentItems[currentPostIndex + 1] %}
```
Comment (matthiasrohmer) [inline site/_includes/partials/post-navigation.njk:14]: There is https://www.11ty.dev/docs/filters/collection-items/#getpreviouscollectionitem-and-getnextcollectionitem - is that of help here?

File: site/_includes/partials/post-navigation.njk [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{% set contentItems = null %}
+{% set contentType = "" %}
+
+{% if page.url | getContentType === 'articles' %}
+  {% set contentItems = collections.articles %}
+{% endif %}
+
+{% if page.url | getContentType === 'blog' %}
+  {% set contentItems = collections.blogPosts %}
+{% endif %}
```
[the remark this answers is not in the document]
Comment (simgobb) [reply site/_includes/partials/post-navigation.njk]: Unfortunately, both blog posts and articles use the same layout. Not a massive fan of checking the url, but couldn't think of anything else

[the remark this answers is not in the document]
Comment (matthiasrohmer) [reply site/_includes/partials/post-navigation.njk]: Let's use 11ty data cascade instead 🙂  create `site/en/blog/blog.11tydata.js` with

```js
module.exports = {
  postType: 'blog',
};
```

Then you should be able to do `postType == 'blog'` in the template, respectively the same for articles.

That's how our locales work. See site/en/en.11tydata.js for example.

File: site/_includes/partials/post-navigation.njk [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{% set contentItems = null %}
```
Comment (matthiasrohmer) [inline site/_includes/partials/post-navigation.njk:1]: This seems to be unused...?

File: site/_includes/partials/post-navigation.njk [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  {% set collectionName = 'articles-' + locale %}
+{% endif %}
+
+{% if type === 'blogPost' %}
+  {% set collectionName = 'blog-' + locale %}
+{% endif %}
+
+{% if collections[collectionName] %}
+  {% set previousPost = collections[collectionName]  | getPreviousCollectionItem(page) %}
+  {% set nextPost = collections[collectionName]  | getNextCollectionItem(page) %}
```
Comment (matthiasrohmer) [inline site/_includes/partials/post-navigation.njk:13]: Nit: double spaces before `|`

File: site/_scss/blocks/_post-navigation.scss [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    content: '';
+    display: inline-block;
+    height: px-to-rem(16px);
+    margin-right: px-to-rem(20px);
+    margin-top: px-to-rem(-2px);
+    vertical-align: middle;
+    width: px-to-rem(16px);
+  }
+
+  &__next &__label::before {
```
Comment (matthiasrohmer) [inline site/_scss/blocks/_post-navigation.scss:72]: Nit: make this `&__label:before` and then only flip in `&_prev &__label:before` with `transform` - saves some bytes in the CSS.

**Review by matthiasrohmer — COMMENTED** [review]
image

On the first article the (empty) button to the left does still have a pointer icon and a hover effect. That also applies to the last one. Consider using the CSS `&:empty` selector to fix this.

**Review by matthiasrohmer — APPROVED** [review]

**Pull request merged**